akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -260,6 +259,18 @@ class Source[+Out, +Mat](delegate: scaladsl.Source[Out, Mat]) extends Graph[Sour
     new Source(delegate.via(flow))
 
   /**
+   * Transform this [[Source]] by appending a Reactive Streams compliant [[org.reactivestreams.Processor]]. This
+   * method takes a factory that instantiates the [[org.reactivestreams.Processor]] for every materialization
+   * of the resulting [[Source]].
+   *
+   * The materialized value of the combined [[Source]] will be the materialized
+   * value of the current source.
+   */
+  def viaProcessor[T](processorFactory: function.Creator[Processor[Out @uncheckedVariance, T]]): Source[T, Mat] = {
+    new Source(delegate.viaProcessor(() ⇒ processorFactory.create()))
+  }
```

Comment:
Please copy the same graph from scaladsl to this one, it helps a lot when learning those things, see: https://github.com/akka/akka/pull/17953/files#diff-a4d5987f74d69093ccf1264366e4bbc1R111